nmcli: Introduce SIT Tunnel

ssahani opened the pull request:
This work enables to add sit tunnel via nmcli module

Signed-off-by: Susant Sahani <susant@redhat.com>

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->

- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
nmcli
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
2.8
```

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```
  - nmcli:
       state: present
       type: sit
       conn_name: sit_test1
       autoconnect: yes
       ip_tunnel_dev: enp0s8
       ip_tunnel_local: 192.168.1.2
       ip_tunnel_remote: 192.168.1.5
```

Akasurde commented on lib/ansible/modules/net_tools/nmcli.py:
```diff
@@ -1066,6 +1089,61 @@ def modify_connection_vlan(self):
         return cmd
 
+    def create_connection_sit(self):
+        cmd = [self.nmcli_bin]
```
```python
cmd = [self.nmcli_bin, 'con', 'add', 'type', 'ip-tunnel', 'mode', 'sit', 'con-name']
```

ssahani replied:
yes sure . updated.

Akasurde commented on lib/ansible/modules/net_tools/nmcli.py:
```diff
@@ -1066,6 +1089,61 @@ def modify_connection_vlan(self):
+        return cmd
+
+    def modify_connection_sit(self):
+        cmd = [self.nmcli_bin]
```
```python
cmd = [self.nmcli_bin, 'con', 'mod']
```

Akasurde reviewed (changes requested):
Thanks for the contribution. Could you please write a test for this change here - https://github.com/ansible/ansible/blob/devel/test/units/modules/net_tools/test_nmcli.py ? Thanks.

ssahani commented:
I have updated not sure how to run these tests locally ?

Akasurde commented:
@ssahani You can run tests locally using following command - 

```
# pytest -r a -n auto --color yes --junit-xml test/results/junit/python2.6-units.xml -v test/units/modules/net_tools/test_nmcli.py
```

ssahani commented:
@Akasurde Thanks for reviewing. Seems all tests passed.

ssahani commented:
@Akasurde  Ping

Akasurde commented:
@ssahani Could you please rebase this ? needs_info

Akasurde commented:
@ssahani #45933 is merged. Could you please rebase this ? Thanks.

ssahani committed:
nmcli: Introduce SIT Tunnel

This work enables to add sit tunnel via nmcli module

Signed-off-by: Susant Sahani susant@redhat.com

SUMMARY
ISSUE TYPE
Feature Pull Request
COMPONENT NAME
nmcli

ANSIBLE VERSION
2.8
ADDITIONAL INFORMATION
  - nmcli:
       state: present
       type: sit
       conn_name: sit_test1
       autoconnect: yes
       ip_tunnel_dev: enp0s8
       ip_tunnel_local: 192.168.1.2
       ip_tunnel_remote: 192.168.1.5

ssahani commented:
@Akasurde Thanks updated.

Akasurde commented:
rebuild_merge
